python_modules/dagster/dagster/_core/definitions/base_asset_graph.py

```diff
@@ -354,6 +354,15 @@ def root_executable_asset_keys(self) -> AbstractSet[AssetKey]:
     @abstractmethod
     def asset_check_keys(self) -> AbstractSet[AssetCheckKey]: ...
 
+    def get_asset_check_keys_for_assets(
```

Comment:
This is O(# asset checks in the deployment) when it could be much less if we kept a deps data structure for asset keys like we do for assets. We should consider adding one before merging this PR.